config/checkstyle/checkstyle.xml

```diff
@@ -78,6 +78,13 @@
         </module>
         -->
 
+        <module name="Regexp">
+            <property name="format" value="new (ActivitiesResourceApi|AllHostsResourceApi|AuditsResourceApi|AuthRoleMetadatasResourceApi|AuthRolesResourceApi|AuthServiceResourceApi|AuthServiceRoleCommandsResourceApi|AuthServiceRoleConfigGroupsResourceApi|AuthServiceRolesResourceApi|BatchResourceApi|CdpResourceApi|CertManagerResourceApi|ClouderaManagerResourceApi|ClustersResourceApi|CmPeersResourceApi|CommandsResourceApi|DashboardsResourceApi|DataContextsResourceApi|EventsResourceApi|ExternalAccountsResourceApi|ExternalUserMappingsResourceApi|HostsResourceApi|HostTemplatesResourceApi|ImpalaQueriesResourceApi|MgmtRoleCommandsResourceApi|MgmtRoleConfigGroupsResourceApi|MgmtRolesResourceApi|MgmtServiceResourceApi|NameservicesResourceApi|ParcelResourceApi|ParcelsResourceApi|ProcessResourceApi|ReplicationsResourceApi|RoleCommandsResourceApi|RoleConfigGroupsResourceApi|RolesResourceApi|ServicesResourceApi|SnapshotsResourceApi|TagsResourceApi|TimeSeriesResourceApi|ToolsResourceApi|UsersResourceApi|WatchedDirResourceApi|YarnApplicationsResourceApi)" />
```

Comment:
I know it's not ideal but actually it's impossible to get these classes in a unit test or a custom checkstyle java rule.